### PR TITLE
Release/1.4.1

### PR DESCRIPTION
**Pull request by tznind**
## Proposed Changes

This is to release 1.4.1 which contains:

- fix for retry batches
- random backoff/retry times for retry batches
- lock on IdentifierMapper producer

## Types of changes

What types of changes does your code introduce? Tick all that apply.

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation-Only Update (if none of the other choices apply)
  - In this case, ensure that the message of the head commit from the source branch is prefixed with `[skip ci]`

## Checklist

By opening this PR, I confirm that I have:

- [x] Reviewed the [contributing](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTING.md) guidelines for this repository
- [x] Ensured that the PR branch is in sync with the target branch (i.e. it is automatically merge-able)
- [x] Updated any relevant API documentation
- [ ] Created or updated any tests if relevant
- [x] Accurately updated the [CHANGELOG](https://github.com/SMI/SmiServices/blob/master/CHANGELOG.md)
  - NOTE: This ***must*** include any changes to any of the following files: default.yaml, any of the RabbitMQ server configurations, GlobalOptions.cs
- [x] Listed myself in the [CONTRIBUTORS](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTORS.md) file 🚀
- [x] Requested a review by one of the repository maintainers

## Issues

If relevant, tag any issues that are *expected* to be resolved with this PR. E.g.:

- Closes #\<issue-number>
- ...

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -147,7 +156,8 @@ First stable release after importing the repository from the private [SMIPlugin]
 - Anonymous `MappingTableName` must now be fully specified to pass validation (e.g. `mydb.mytbl`).  Previously skipping database portion was supported.
 
 
-[Unreleased]: https://github.com/SMI/SmiServices/compare/v1.4.0...develop
+[Unreleased]: https://github.com/SMI/SmiServices/compare/v1.4.1...develop
+[1.4.0]:  https://github.com/SMI/SmiServices/compare/v1.4.0...v1.4.1
```
Comment (rkm) [inline CHANGELOG.md]: Should be `1.4.1`

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,15 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ...
```
Comment (rkm) [inline CHANGELOG.md]: Add 1.4.1 heading

**Comment by rkm** [comment]
Looks good other than the documentation comments

File: CHANGELOG.md (hunk 2 above)
Comment (tznind) [reply CHANGELOG.md]: Done

**Comment by rkm** [comment]
The Java build is still erroring, but feel free to merge if the C# job passes

**Pull request merged**